lib/Parse/Lexer.cpp

```diff
@@ -30,17 +30,19 @@
 // FIXME: Figure out if this can be migrated to LLVM.
 #include "clang/Basic/CharInfo.h"
 
+#include <limits>
+
+using namespace swift;
+using namespace swift::syntax;
+
 // Regex parser delivered via libSwift
-#include "swift/Parse/ExperimentalRegexBridging.h"
 static ParseRegexStrawperson parseRegexStrawperson = nullptr;
 void Parser_registerParseRegexStrawperson(ParseRegexStrawperson fn) {
   parseRegexStrawperson = fn;
 }
-
-#include <limits>
-
-using namespace swift;
-using namespace swift::syntax;
+ParseRegexStrawperson Lexer::getExperimentalRegexParsingFn() {
+  return parseRegexStrawperson;
+}
```

Comment:
At this point, this function and `Parser_registerParseRegexStrawperson` do nothing to do with `Lexer`. Can we move these to `Parser`. I guess we can implement this in `Parser.cpp` or maybe introduce `ParseRegex.cpp`.

Reply:
@rintaro I've moved the regex lexing and parsing logic into a ParseRegex.cpp file, that way we don't have to expose a method to grab the parsing function, does that look good to you?